client/landing/stepper/declarative-flow/import-flow.ts

```diff
@@ -168,16 +178,42 @@ const importFlow: Flow = {
 				}
 
 				case 'sitePicker': {
-					const newQueryParams =
-						( providedDependencies?.queryParams as { [ key: string ]: string } ) || {};
+					switch ( providedDependencies?.action ) {
+						case 'update-query': {
+							const newQueryParams =
+								( providedDependencies?.queryParams as { [ key: string ]: string } ) || {};
+
+							Object.keys( newQueryParams ).forEach( ( key ) => {
+								newQueryParams[ key ]
+									? urlQueryParams.set( key, newQueryParams[ key ] )
+									: urlQueryParams.delete( key );
+							} );
+
+							return navigate( `sitePicker?${ urlQueryParams.toString() }` );
+						}
+
+						case 'select-site': {
+							const selectedSite = providedDependencies.site as SiteExcerptData;
 
-					Object.keys( newQueryParams ).forEach( ( key ) => {
-						newQueryParams[ key ]
-							? urlQueryParams.set( key, newQueryParams[ key ] )
-							: urlQueryParams.delete( key );
-					} );
+							if ( selectedSite && sourceSite ) {
```

Comment:
Just a heads up this will soon be moved and handled in the component itself. `useSiteQuery` is not working well with sites that don't have the Jetpack installed. To do so, we have a new endpoint for this Automattic/dotcom-forge#2255, but it requires both the target site id and source site slug. I'll be handling this in this task: #77234